src/safeds_runner/server/messages.py

```diff
@@ -324,9 +438,23 @@ def validate_placeholder_query_message_data(message_data: dict[str, Any] | str)
 
     Returns
     -------
-    tuple[str | None, str | None]
-        A tuple containing either a validated message data as a string or an error message.
+    tuple[MessageQueryInformation | None, str | None]
+        A tuple containing either the validated message data or an error message.
     """
-    if not isinstance(message_data, str):
-        return None, "Message data is not a string"
-    return message_data, None
+    if not isinstance(message_data, dict):
+        return None, "Message data is not a JSON object"
+    if "name" not in message_data:
+        return None, "No 'name' parameter given"
+    if (
+        "window" in message_data
+        and "begin" in message_data["window"]
+        and not isinstance(message_data["window"]["begin"], int)
+    ):
+        return None, "Invalid 'window'.'begin' parameter given"
+    if (
+        "window" in message_data
+        and "size" in message_data["window"]
+        and not isinstance(message_data["window"]["size"], int)
+    ):
+        return None, "Invalid 'window'.'size' parameter given"
+    return MessageQueryInformation.from_dict(message_data), None
```

Comment:
While semantically equivalent here, I prefer one if/elif/else statement instead of a sequence of if statements if there are many branches. This indicates at a glance that exactly one branch will be executed and no fallthrough occurs.